.github/workflows/flowzone.yml

```diff
@@ -490,21 +490,47 @@ jobs:
     name: NPM draft
     runs-on: ubuntu-latest
     needs: [npm_matrix, source_check]
-    if:
-      needs.source_check.outputs.npm_private != 'true' &&
-      inputs.npmjs_repository != ''
+    if: inputs.npmjs_repository != ''
 
     defaults:
       run:
-        working-directory: .
+        working-directory: ${{ inputs.working_directory }}
         shell: bash
 
     steps:
+      - name: Download versioned source
+        uses: actions/download-artifact@v3
+        with:
+          name: versioned-source-${{ github.event.pull_request.head.sha }}
+
+      - name: Extract versioned source
+        working-directory: .
+        run: tar -xvf versioned-source.tar
+
       - name: Publish draft tag
         run: |
-          echo "not yet implemented!"
+          echo '//${{ inputs.npmjs_repository }}/:_authToken=${{ secrets.NPM_TOKEN }}' > ~/.npmrc
+
+          npm whoami
+
+          branch="$(echo '${{ github.event.pull_request.head.ref }}' | sed 's/\//-/g')"
+
+          version="$(cat < ../VERSION)"
```

Comment:
Seems we should get this from package.json if possible

Reply:
We also don't know that VERSION will be up one relative dir, definitely use `jq -r '.version' package.json`